lightning/src/chain/channelmonitor.rs

```diff
@@ -1907,9 +1907,11 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 						continue;
 					}
 				} else if htlc.0.cltv_expiry > self.best_block.height() + 1 {
-					// Don't broadcast HTLC-Timeout transactions immediately as they dont' meet the
-					// current locktime requirements on-chain. We will broadcast them when the
-					// commitment transaction confirms automatically.
+					// Don't broadcast HTLC-Timeout transactions immediately as they don't meet the
+					// current locktime requirements on-chain. We will broadcast them in
+					// `block_confirmed` when `would_broadcast_at_height` returns true.
+					// Note that we add + 1 as transactions are broadcastable when they can be
+					// confirmed in the next block, not the current height.
```

Comment:
Right, its just noting that we can always broadcast when the locktime is `X + 1`. I can drop if you think its obvious but Antoine suggested it.

* Fun fact of the day - Bitcoin Core had a bug for many years where it actually wouldn't accept-for-relay transactions until they were confirm-able in the *current* block, not X+1.

Reply:
Right, if the current height is X, then the next block which will be mined is X + 1. Thus, we can always broadcast something with a CLTV expiry of X + 1 as it can go into the next block (but could not appear in the current best block).

Reply:
Ah, that makes sense. I think dropping "not the current height" would be beneficial as including it can lead to confusion.